File: app/src/main/java/com/daniel/mobilepauker2/dropbox/NetworkStateReceiver.java
package com.daniel.mobilepauker2.dropbox;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.daniel.mobilepauker2.utils.Log;

public class NetworkStateReceiver extends BroadcastReceiver {
    private static final String TAG = "NetworkStateReceiver";
    private ReceiverCallback callback;

    public NetworkStateReceiver(ReceiverCallback callback) {
        super();
        this.callback = callback;
    }

    @Override
    public void onReceive(final Context context, final Intent intent) {

        Log.d(TAG, "Network connectivity change");

        if (intent.getExtras() != null) {
            final ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            if (cm != null) {
                final NetworkInfo ni = cm.getActiveNetworkInfo();

                if (ni != null && ni.isConnectedOrConnecting()) {
                    Log.i(TAG, "Network " + ni.getTypeName() + " connected");
                } else if (intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, Boolean.FALSE)) {
                    Log.d(TAG, "There's no network connectivity");
                    callback.connectionLost();
                }
            }
        }
    }

    public interface ReceiverCallback {
        void connectionLost();
    }
}

File: app/src/main/java/com/daniel/mobilepauker2/dropbox/NetworkStateReceiver.kt
package com.daniel.mobilepauker2.dropbox

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import com.daniel.mobilepauker2.pauker_native.Log

class NetworkStateReceiver(private val callback: ReceiverCallback) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.d(
            TAG,
            "Network connectivity change"
        )
        if (intent.extras != null) {
            val cm =
                context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val ni = cm.activeNetworkInfo
            if (ni != null && ni.isConnectedOrConnecting) {
                Log.i(
                    TAG,
                    "Network " + ni.typeName + " connected"
                )
            } else if (intent.getBooleanExtra(
                    ConnectivityManager.EXTRA_NO_CONNECTIVITY,
                    false
                )
            ) {
                Log.d(
                    TAG,
                    "There's no network connectivity"
                )
                callback.connectionLost()
            }
        }
    }

    interface ReceiverCallback {
        fun connectionLost()
    }

    companion object {
        private const val TAG = "NetworkStateReceiver"
    }

}